### DISCUSSION TITLE
Error: Unable to find any supported Python versions.

### DISCUSSION THREAD

**Discussion by IK-R-S** [vercel/community#6287] [Bug Report]
### Page to Investigate

https://vercel.com/templates/python/flask-hello-world

### Steps to Reproduce

![image](https://github.com/vercel/community/assets/73291742/fa06a2bb-a4c8-46e7-8943-acc7a6bf9e3d)

I recently tried to deploy an application using Vercel's Flask template and the following error occurred:

`"Error: Unable to find any supported Python versions."`

I tried redeploying for 2 days in a row and out of curiosity, testing the python-hello-world template as well, which generated the same deploy error. What could it be ?

![image](https://github.com/vercel/community/assets/73291742/1bf6d779-e37b-4220-a239-1064b74ecd88)

I have seen previous reports that say about incompatibility with node 20.x, but the project is already configured with node 18.x

**Comment by xavierace990** [comment]
is there any way to do this

**Comment by amyegan** [comment]
Thanks for reporting this! I have something you to try that worked for me. Go into project settings and change the Node.js version to 18.x, then redeploy or start a new deployment. Does that fix it?

**Reply by IK-R-S** [reply]
**Editing "node" to 18.x in package.json**

```json
{
  "engines": {
    "node": "18.x"
  }
}
```

editing the node version even if it is already pre-configured as 18.x seems to solve the problem, the strange thing is that I tried it and the first attempt it didn't work, while trying again today the app completed the build.

**Reply by amyegan** [reply]
Yeah, this looks like a bug. Thanks for giving that a try, and I'm glad it got your project unstuck.

The build engineers are working on a fix now.

**Reply by Opeoluwa-Fatunmbi** [reply]
@amyegan I am facing the same issue trying to host my django project
![Screenshot (622)](https://github.com/vercel/community/assets/101920613/dbb992ae-27b1-45f7-b141-f30272f1afbe)

**Reply by IK-R-S** [reply]
> @amyegan I am facing the same issue trying to host my django project
> ![Screenshot (622)](https://github.com/vercel/community/assets/101920613/dbb992ae-27b1-45f7-b141-f30272f1afbe)
> 

Try to edit node version to 18.x even it's already configured as 18.x

Then rebuild the app.

**Reply by amyegan** [reply]
Make sure you [change the Node version in Project Settings](https://vercel.com/docs/functions/runtimes/node-js#node.js-version) rather than in the `vercel.json` config 🙏

**Comment by r-rajababushah** [comment]
It doesn't work even if i changed the node version 18.x to 18.x again

**Reply by IK-R-S** [reply]
See if the change is on production

**Reply by r-rajababushah** [reply]
When i changed node version in vercel settings , it works

**Reply by satyajitgiram** [reply]
changing the version from the project settings (in vercel's deployeed project) ( from 20 to 18 ), worked for me

**Comment by see7e** [comment]
Same issu here tried buiding either with [Flask](https://github.com/see7e/YourFinance/deployments) or [Django](https://github.com/see7e/django_company-modules/deployments) but got the same problem:

```
Running "vercel build"
--
12:56:33.887 | Vercel CLI 33.6.2
12:56:34.020 | WARN! Due to `builds` existing in your configuration file, the Build and Development Settings defined in your Project Settings will not apply. Learn More: https://vercel.link/unused-build-settings
12:56:34.041 | Error: Unable to find any supported Python versions.
12:56:34.041 | Learn More: http://vercel.link/python-version
```


Here is the current `vercel.json`config file:

```json
{
  "version": 2,
  "builds": [
    {
      "src": "default/settings/wsgi.py",
      "use": "@vercel/python",
      "config": { "maxLambdaSize": "15mb", "runtime": "python3.9" }
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "default/settings/wsgi.py"
    }
  ]
}

```

**Reply by r-rajababushah** [reply]
inside vercel , you can change node js version to make to work

**Reply by see7e** [reply]
My man, sorry for the late response (tough week), the Node version was on 20.x and i didn't knew that don supports Python. Thanks for the heads up!

**Comment by mchrisman** [comment]
what if you're not using a framework, just a bare python script?

**Reply by see7e** [reply]
What kind of script? It has any dependency on newer versions?

**Reply by amyegan** [reply]
Even with just Python, please set the Node.js version to `18.x` in project settings 🙏

**Comment by tesaabi** [comment]
Hii please helpp

it work for me when i changed the node version to 18, but then when i want to deploy it on production, it resulted this error in the cmd
"Due to `builds` existing in your configuration file, the Build and Development Settings defined in your Project Settings will not apply. Learn More: https://vercel.link/unuused-build-settings"

and also this error on the web
This Serverless Function has crashed.

Your connection is working correctly.

Vercel is working correctly.

500: INTERNAL_SERVER_ERROR
Code: FUNCTION_INVOCATION_FAILED

on the log i got this
LAMBDA_WARNING: Unhandled exception. The most likely cause is an issue in the function code. However, in rare cases, a Lambda runtime update can cause unexpected function behavior. For functions using managed runtimes, runtime updates can be triggered by a function change, or can be applied automatically. To determine if the runtime has been updated, check the runtime version in the INIT_START log entry. If this error correlates with a change in the runtime version, you may be able to mitigate this error by temporarily rolling back to the previous runtime version. For more information, see https://docs.aws.amazon.com/lambda/latest/dg/runtimes-update.html
[ERROR] Runtime.ImportModuleError: Unable to import module 'vc__handler__python': No module named 'flask'
Traceback (most recent call last):

**Reply by amyegan** [reply]
There is a build container bug that is being addressed, and the available workaround is to go into project settings and change Node.js Version to `18.x`. However, what you reported seems like a different problem. 

Please [start a new discussion](https://github.com/vercel/community/discussions/new) with more information about your project. It's good to include a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) if possible 🙏

**Comment by xavierace990** [comment]
Vercel is no longer hosting python
They're just acting like they can

**Reply by tesaabi** [reply]
do u know any other web host for flask?

**Reply by amyegan** [reply]
Vercel still has a [Python runtime](https://vercel.com/docs/functions/runtimes/python) that is currently in beta and available on all plans. But you are always free to choose the tools that make most sense for you and your project.

I don't have a list of recommended alternatives, but I found this post on Reddit that might help you: https://www.reddit.com/r/learnpython/comments/14jimd7/best_webhosting_for_python/

**Reply by tesaabi** [reply]
thankyou i'll check it later
